file_organizer_v2/src/file_organizer/methodologies/para/detection/heuristics.py

```
@@ -0,0 +1,450 @@
+"""
+PARA Heuristics Engine
+
+Multi-factor heuristic detection system for automatic PARA categorization.
+Uses temporal, content, structural, and AI-based heuristics.
+"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
```

Ruff (0.14.13)

`typing.Dict` is deprecated, use `dict` instead

(UP035)